### PR TITLE
[PORT] Removes rad immunity trait from IPCs and gives them NSV's radiation interaction

### PR DESCRIPTION
**Pull request by YakiAttaki**
## About The Pull Request
Ports [#1144](https://github.com/BeeStation/NSV13/pull/1144) from NSV
for whatever reason, IPCs are the only "silicon" type mob that are immune to rads. Not even borgs, drones, or AIs are immune. Why the hell are IPCs rad immune? Kinda dumb.

## Why It's Good For The Game
Means IPCs can't just tank massive radiation and shrug it off. Makes more sense mechanically as they're just borgs with hands and more freedom.

## Changelog
:cl:
port: adds IPC radiation interaction from NSV
balance: removed IPC's rad-immune trait
/:cl:

### CODE REVIEW

**Comment by ToasterBan** [comment]
borgs and drones are rad immune??

**Comment by heepox** [comment]
Borgs and drones and AI do not take damage from radiation even if they do not have TRAIT_RADIMMUNE as they literally cannot take toxin damage.

**Comment by YakiAttaki** [comment]
radiation still kills borgs and drones.

**Comment by ToasterBan** [comment]
> radiation still kills borgs and drones.

it litterly doesn't?

**Comment by YakiAttaki** [comment]
> > radiation still kills borgs and drones.
> 
> it litterly doesn't?

i literally watched a borg die due to radiation

**Comment by ToasterBan** [comment]
> > > radiation still kills borgs and drones.
> > 
> > 
> > it litterly doesn't?
> 
> i literally watched a borg die due to radiation

well then you watched wrong

**Comment by amidoingitright** [comment]
to clarify borgs and ai aswell as drones suffer burn damage
this occurs at fusion tier rads
which IPC's are immune to

**Comment by utilbird** [comment]
> well then you watched wrong

They didn't, rads in large amounts causes burn, usually once it's at that stage it one shots them once they're contaminated

**Comment by ToasterBan** [comment]
did you ever do any testing? or do you just want to nerf ipcs for no reason, because 
1. silicons are immune to toxin damage so radiation is harmless
2. ipcs are stupidly hard to deal toxin damage with

**Comment by amidoingitright** [comment]
there have been fusion rads powerful enough to kill the AI in its sat with the burn damage
are you even listening?

**Comment by ToasterBan** [comment]
> there have been fusion rads powerful enough to kill the AI in its sat with the burn damage
> are you even listening?

rads rarely are that high, and at that point just let ipcs have the radiation immunity because ipcs need some form off buff since they are litteral tissue paper

**Comment by ToasterBan** [comment]
> there have been fusion rads powerful enough to kill the AI in its sat with the burn damage
> are you even listening?

that was pre-nerf fusion, for that to happen you need to make fusion right next to the ai

**Comment by amidoingitright** [comment]
IPC's are also immune to organ decay
have the easiest revival steps
have their brain in their chest {wildcard but it can trick people}

**Comment by utilbird** [comment]
> that was pre-nerf fusion, for that to happen you need to make fusion right next to the ai

It happened 40 minutes ago. You're starting to hurt my brain, please stop throwing random false facts around to prove your point.

**Comment by ToasterBan** [comment]
> IPC's are also immune to organ decay
> have the easiest revival steps
> have their brain in their chest {wildcard but it can trick people}

they get 3 shot by welders and can be made completely unrevivable by removing their brain , also emps cause them to just get destroyed, something a tider can easily acquire

they have many ups and man downsides and rad immunity is one of them

**Comment by ToasterBan** [comment]
> > that was pre-nerf fusion, for that to happen you need to make fusion right next to the ai
> 
> It happened 40 minutes ago. You're starting to hurt my brain, please stop throwing random false facts around to prove your point.

I don't know how you manage to get radiation so strong you kill the ai at sat so beats me

**Comment by ToasterBan** [comment]
also spiecies changes require maintainer approval

**Comment by amidoingitright** [comment]
> also spiecies changes require maintainer approval

> thefakeelon is a bee maintainer

why are you defending a moot point?

**Comment by utilbird** [comment]
@amidoingitright 
No, he's right. @YakiAttaki should have asked before PRing

**Review by utilbird — DISMISSED** [review]
nonetheless

**Comment by YakiAttaki** [comment]
My apologies

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Look at porting the stuff on NSV that makes IPCs take brain damage instead of mutating when exposed to rads.

**Comment by PowerfulBacon** [comment]
Also would need maintainer approval for species changes, but since you have an approval anyway it wont be too hard probably

**Comment by Archanial** [comment]
> 
> 
> > radiation still kills borgs and drones.
> 
> it litterly doesn't?

Extreme rads kill borgs. You are wrong.

**Comment by utilbird** [comment]
If you're porting then you need to credit the original authors

File: code/datums/weather/weather_types/radiation_storm.dm
Comment (utilbird) [inline code/datums/weather/weather_types/radiation_storm.dm:57]: what in the goddamn. what is this semicolom voodoo

File: code/datums/weather/weather_types/radiation_storm.dm
Comment (utilbird) [inline code/datums/weather/weather_types/radiation_storm.dm:45]: I don't see why you'd want to use traits for this, you should just be able to use `isipc(L)`

File: code/modules/mob/living/carbon/human/species.dm
Comment (utilbird) [inline code/modules/mob/living/carbon/human/species.dm:1186]:
```suggestion
		var/obj/item/organ/stomach/S = H.getorganslot(ORGAN_SLOT_STOMACH)
		if(S && istype(S) !(S.organ_flags & ORGAN_SYNTHETIC)) // synthetics can't vomit // synthetics can't vomit
```
organs flags are bit flags so you'll have to use bit operators.

File: code/modules/mob/living/carbon/life.dm
Comment (utilbird) [inline code/modules/mob/living/carbon/life.dm:383]: This is already checked in `adjustToxLoss`

File: code/modules/reagents/chemistry/recipes/medicine.dm
Comment (utilbird) [inline code/modules/reagents/chemistry/recipes/medicine.dm:302]: The name is a bit misleading 😆

File: code/modules/mob/living/carbon/human/species.dm
Comment (utilbird) [inline code/modules/mob/living/carbon/human/species.dm:1215]: emote calls should (almost) always be done async for.. reasons
```suggestion
			INVOKE_ASYNC(H, /mob.proc/emote, emote_type)
```

File: code/datums/weather/weather_types/radiation_storm.dm
Comment (utilbird) [inline code/datums/weather/weather_types/radiation_storm.dm:66]:
```suggestion
							INVOKE_ASYNC(H, /mob.proc/emote, emote_type)
```

File: code/__DEFINES/traits.dm
Comment (utilbird) [inline code/__DEFINES/traits.dm:108]: Just check for IPC brains

**Review by utilbird — CHANGES_REQUESTED** [review]

**Comment by Lagomorphica** [comment]
Seems fine enough, however, ,make sure the brain traumas it gives you are only Basic, as was highlighted in the PR on NSV - making it only be ones curable with chemistry makes far more sense, since thats about equal to how annoying must mutations are to a human. Randomly getting a Deep Rooted or lobotomy-required brain trauma is a lot worse than almost any mutation I can think of, since even Acid Skin just requires you to be revived and given a mutadone pill.

People do seem to think IPCs are a bit overpowered at the moment, and removing some of their utility that plasmamen have might help - even if I am a bit biased towards playing IPC. It will also give Solder more use, albeit I do hope that the amount of brain damage it causes isn't overboard. I would, however, advise spawning some in Engineering roundstart, akin to how normal Engineers get it - which'd require a mapping change. I'm not sure if this'll be the best on Beestation myself (especially if the surgical traumas are needed, since chemists on golden and sage can be quite uncooperative to an IPC of all people asking for medication), but its still worth trying.

As for the E+015 rads thing, I believe that was changed so that 1000 rads does burn damage? However, for both sides, I feel that,

- If it wasn't changed and you still need E+015 rads, it'd be a bit stupid to have a debilitating thing on IPC's - albeit when I looked in radiation code it was at 1000. 
- If it was changed then brain damage would work, albeit I'd say anything above a basic (chem curable) brain trauma is extremely overkill.

Yes, I am biased towards IPCs, but even people on NSV brought this up - having severe or debilitating brain traumas that kill you and are nigh-incurable short of giving yourself -another- brain trauma, in comparison to being able to just pop mutadone, is pretty debilitating.

**Comment by YakiAttaki** [comment]
aight i'll fix this pr up another day

**Pull request closed without merging**
